dev/how-to/maintain/tidb-in-kubernetes/monitoring.md

```diff
@@ -0,0 +1,96 @@
+
+在 [TiDB 集群监控](/how-to/monitor/monitor-a-cluster.md)中有一些监控系统配置的细节可供参考。
+
+### 查看监控面板
+
+可以通过 `kubectl port-forward` 查看监控面板：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl port-forward -n ${namespace} svc/${releaseName}-grafana 3000:3000 &>/tmp/portforward-grafana.log
+```
+
+然后在浏览器中打开 [http://localhost:3000](http://localhost:3000)，默认用户名和密码都为 `admin`。
+
+Grafana 服务默认通过 `NodePort` 暴露，如果 Kubernetes 集群支持负载均衡器，你可以在 `values.yaml` 中将 `monitor.grafana.service.type` 修改为 `LoadBalancer`，然后在执行 `helm upgrade` 后通过负载均衡器访问面板。
+
+如果不需要使用 Grafana，可以在部署时在 `values.yaml` 中将 `monitor.grafana.create` 设置为 `false` 来节省资源。这一情况下需要使用其他已有或新部署的数据可视化工具直接访问监控数据来完成可视化。
+
+### 访问监控数据
+
+对于需要直接访问监控数据的情况，可以通过 `kubectl port-forward` 来访问 Prometheus：
+
+{{< copyable "shell-regular" >}}
+
+```shell
+kubectl port-forward -n ${namespace} svc/${releaseName}-prometheus 9090:9090 &>/tmp/portforward-prometheus.log
+```
+
+然后在浏览器中打开 [http://localhost:9090](http://localhost:9090)，或通过客户端工具访问此地址即可。
+
+Prometheus 服务默认通过 `NodePort` 暴露，如果 Kubernetes 集群支持负载均衡器，你可以在 `values.yaml` 中将 `monitor.prometheus.service.type` 修改为 `LoadBalancer`，然后在执行 `helm upgrade` 后通过负载均衡器访问监控数据。
+
+## Kubernetes 的监控
+
+随集群部署的 TiDB 监控只关注 TiDB 本身各组件的运行情况，并不包括对容器资源、宿主机、Kubernetes 组件和 TiDB Operator 等的监控。对于这些组件或资源的监控，需要在整个 Kubernetes 集群维度部署监控系统来实现。
+
+### 宿主机监控
+
+对宿主机及其资源的监控与传统的服务器物理资源监控相同。
+
+如果在你的现有基础设施中已经有针对物理服务器的监控系统，只需要通过常规方法将 Kubernetes 所在的宿主机添加到现有监控系统中即可；如果没有可用的监控系统，或者希望部署一套独立的监控系统用于监控 Kubernetes 所在的宿主机，也可以使用你熟悉的任意监控系统。
+
+新部署的监控系统可以运行于独立的服务器、直接运行于 Kubernetes 所在的宿主机，或运行于 Kubernetes 集群内，不同部署方式除在安装配置与资源利用上存在少许差异，在使用上并没有重大区别。
+
+常见的可用于监控服务器资源的开源监控系统有：
+ - [CollectD](https://collectd.org/)
+ - [Nagios](https://www.nagios.org/)
+ - [Prometheus](http://prometheus.io/) & [node_exporter](https://github.com/prometheus/node_exporter)
+ - [Zabbix](https://www.zabbix.com/)
+
+一些云服务商或专门的性能监控服务提供商也有各自的免费或收费的监控解决方案可以选择。
+
+我们推荐通过 [Prometheus Operator](https://github.com/coreos/prometheus-operator) 在 Kubernetes 集群内部署基于 [Node Exporter](https://github.com/prometheus/node_exporter) 和 Prometheus 的宿主机监控系统，这一方案同时可以兼容并用于 Kubernetes 自身组件的监控。
+
+### Kubernetes 组件监控
+
+对 Kubernetes 组件的监控可以参考[官方文档](https://kubernetes.io/docs/tasks/debug-application-cluster/resource-usage-monitoring/)提供的方案，也可以使用其他兼容 Kubernetes 的监控系统来进行。
+
+一些云服务商可能提供了自己的 Kubernetes 组件监控方案，一些专门的性能监控服务商也有各自的 Kubernetes 集成方案可以选择。
+
+由于 TiDB Operator 实际上是运行于 Kubernetes 中的容器，选择任一可以覆盖对 Kubernetes 容器状态及资源进行监控的监控系统即可覆盖对 TiDB Operator 的监控，无需再额外部署监控组件。
+
+我们推荐通过 [Prometheus Operator](https://github.com/coreos/prometheus-operator) 部署基于 [Node Exporter](https://github.com/prometheus/node_exporter) 和 Prometheus 的宿主机监控系统，这一方案同时可以兼容并用于对宿主机资源的监控。
+
+## 报警配置
+### TiDB 集群报警
+
+在随 TiDB 集群部署 Prometheus 时，会自动导入一些默认的报警规则，可以通过浏览器访问 Prometheus 的 Alerts 页面查看当前系统中的所有报警规则和状态。
+
+如果需要修改或新增报警规则，可编辑 `charts/tidb-cluster/templates/config/_alert-rules-config.tpl` 文件进行设置，新的配置将在执行 `helm upgrade` 后生效。
```

Comment:
We now request users to use charts from charts.pingcap.com instead of the charts files in tidb-operator repo, how could they get the tpl?

Reply:
Then what should an user do to modify alert rules when they don't have access to the config template?